lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -53,12 +53,16 @@ def available_version
     version_field.try(:fetch_path, 'field', 'default_value')
   end
 
-  def self.import_git_repo(domain_name, git_repo, ref = DEFAULT_BRANCH, ref_type = BRANCH)
-    MiqAeDomain.find_by(:name => domain_name).try(:destroy)
-    MiqAeImport.new(domain_name, import_options(git_repo, ref, ref_type)).import
-    domain = MiqAeDomain.find_by(:name => domain_name)
-    raise MiqAeException::DomainNotFound, "Domain #{domain_name} not found after import" unless domain
-    domain.update_git_info(git_repo, ref, ref_type)
+  def self.import_git_repo(options)
```

Comment:
I don't know what our 'official' stance is on condensing method arguments, but I'm not a huge fan. I think it masks an issue where if the method takes too many arguments, it should arguably either be in a different place that doesn't need as many, or could be broken up into different places so that not as many are needed. It also doesn't let you see at a glance what kind of things need to be passed into the method, and which are optional or not. Just my thoughts though, I don't think this *has* to change.